### PR TITLE
unit test pkg/datasource/formatters/json

### PR DESCRIPTION
**Pull request by upsaurav12**
# [Test: Unit Test For pkg/datasource/formatters/json ]
Ref #3835
This PR add unit test for ``option.go``under ``pkg/datasource/formatters/json``. Covering ``withFields`` ``withShowAll`` ``withPretty`` ``withArray``.

## How to use
This will give you brief idea about the test.
```
cd pkg/datasource/formatters/json
go test ./... -coverprofile=c.out  -v 
```

## Testing done
```
=== RUN   TestWithFields
=== RUN   TestWithFields/Testing_WithFields_with_custom_values
--- PASS: TestWithFields (0.00s)
    --- PASS: TestWithFields/Testing_WithFields_with_custom_values (0.00s)
=== RUN   TestWithShowAll
=== RUN   TestWithShowAll/Testing_WithShowAll_with_custom_values
--- PASS: TestWithShowAll (0.00s)
    --- PASS: TestWithShowAll/Testing_WithShowAll_with_custom_values (0.00s)
=== RUN   TestWithPretty
=== RUN   TestWithPretty/Testing_WithPretty_with_custom_values
--- PASS: TestWithPretty (0.00s)
    --- PASS: TestWithPretty/Testing_WithPretty_with_custom_values (0.00s)
=== RUN   TestWithArray
=== RUN   TestWithArray/Testing_WithArray_with_custom_values
--- PASS: TestWithArray (0.00s)
    --- PASS: TestWithArray/Testing_WithArray_with_custom_values (0.00s)
PASS
coverage: 4.2% of statements
ok      github.com/inspektor-gadget/inspektor-gadget/pkg/datasource/formatters/json     0.004s  coverage: 4.2% of statements
```

### CODE REVIEW

**Comment by upsaurav12** [comment]
@burak-ok PTAL

**Review by burak-ok — CHANGES_REQUESTED** [review]
This gets the code coverage of the `options.go` file to 100%, but it doesn't seem to test anything.

I changed some values in `options.go` and the test is still successfull

**Comment by upsaurav12** [comment]
@burak-ok i have changed the tests case. PTAL?

File: pkg/datasource/formatters/json/options_test.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+package json
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestWithFields(t *testing.T) {
```
Comment (burak-ok) [inline pkg/datasource/formatters/json/options_test.go:23]: I would additionally add a test case (or modify one) to check if `useDefault` stays `true` when `fields == nil`

File: pkg/datasource/formatters/json/options_test.go [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+				useDefault: false,
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			actual := WithFields(test.fields)
+			actual(test.formatter)
+			fmt.Println(test.formatter.useDefault)
```
Comment (burak-ok) [inline pkg/datasource/formatters/json/options_test.go:71]: I believe this was used for debugging.
Also the test/assert is missing here

**Review by burak-ok — APPROVED** [review]
Thanks for working on this
